### PR TITLE
#85 migrate circle to ellipse + refactor to ts

### PR DESCRIPTION
**Pull request by ensemenova**

### CODE REVIEW

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+    restruct.markItemRemoved()
+    restruct.clearVisel(restruct.simpleObjects.get(this.data.id).visel)
+    restruct.simpleObjects.delete(this.data.id)
+
+    struct.simpleObjects.delete(this.data.id)
+  }
+
+  invert(): Base {
+    const ret = new SimpleObjectAdd()
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:110]: please rename `ret` to `simpleObject`

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    ret.data = this.data as any
+    return ret
+  }
+}
+
+class SimpleObjectMove extends Base {
+  data: {
+    id: string
+    d: any
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:119]: what is it?

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+      this.data.d,
+      this.data.noinvalidate
+    )
+  }
+}
+
+class SimpleObjectResize extends Base {
+  data: {
+    id: string
+    d: any
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:152]: same

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  }
+
+  execute(restruct: any): void {
+    const struct = restruct.molecule
+    const id = this.data.id
+    const d = this.data.d
+    const current = this.data.current
+    const item = struct.simpleObjects.get(id)
+    const anchor = this.data.anchor
+    if (item.mode === SimpleObjectMode.ellipse) {
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:178]: too much if-statements

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 5]
```diff
@@ -0,0 +1,291 @@
+import Vec2 from '../../util/vec2'
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:1]: add copyright

File: packages/ketcher-react/src/script/chem/struct/simpleObject.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import Vec2 from '../../util/vec2'
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/chem/struct/simpleObject.ts:1]: add copyright

File: packages/ketcher-react/src/script/format/chemGraph/fromGraph/simpleObjectToStruct.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+/****************************************************************************
+ * Copyright 2020 EPAM Systems
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/format/chemGraph/fromGraph/simpleObjectToStruct.ts:2]: update year

File: packages/ketcher-react/src/script/format/chemGraph/fromGraph/simpleObjectToStruct.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+import { SimpleObject, SimpleObjectMode } from '../../../chem/struct'
+import Vec2 from '../../../util/vec2'
+
+export function simpleObjectToStruct(graphItem, struct) {
+  let object =
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/format/chemGraph/fromGraph/simpleObjectToStruct.ts:20]: const

**Review by TomasLight — CHANGES_REQUESTED** [review]

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+    restruct.simpleObjects.set(
+      this.data.id,
+      new ReSimpleObject(struct.simpleObjects.get(this.data.id))
+    )
+
+    const positions = [...this.data.pos]
+    if (this.data.toCircle) {
+      positions[1] = makeCircleFromEllipse(positions[0], positions[1])
+    }
+    struct.simpleObjectSetPos(
```
[the remark this answers is not in the document]
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts:52]: This method has 2 params id and pos. Id for selecting exact simple object from all and pos to set

File: packages/ketcher-react/src/script/editor/actions/simpleobject.ts [hunk 10]
```diff
@@ -22,14 +22,23 @@ export function fromSimpleObjectDeletion(restruct, id) {
...
-  action.addOp(new op.SimpleObjectAdd(pos, mode))
+  action.addOp(new op.SimpleObjectAdd(pos, mode, toCircle))
   return action.perform(restruct)
 }
 
-export function fromSimpleObjectResizing(restruct, id, d, current, anchor) {
+export function fromSimpleObjectResizing(
+  restruct,
+  id,
+  d,
```
[the remark this answers is not in the document]
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/actions/simpleobject.ts:34]: not very clearly, it's some kind of diff in position

File: packages/ketcher-react/src/script/chem/struct/simpleObject.ts (hunk 6 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/chem/struct/simpleObject.ts:16]: done

**Pull request closed without merging**